src/async-computed.ts

```diff
@@ -0,0 +1,205 @@
+import { Signal } from "signal-polyfill";
+
+export type AsyncComputedState = "initial" | "pending" | "complete" | "error";
+
+export interface AsyncComputedOptions<T> {
+  /**
+   * The initial value of the AsyncComputed.
+   */
+  initialValue?: T;
+}
+
+/**
+ * A signal-like object that represents an asynchronous computation.
+ *
+ * AsyncComputed takes a compute function that performs an asynchronous
+ * computation and runs it inside a computed signal, while tracking the state of
+ * the computation, including its most recent completion value and error.
+ *
+ * Compute functions are run when the `state`, `value`, `error`, or `complete`
+ * properties are read, and are re-run when any signals that they read change.
+ *
+ * If a new run of the compute function is started before the previous run has
+ * completed, the previous run will have its AbortSignal aborted, and the
+ * result of the previous run will be ignored.
+ */
+export class AsyncComputed<T> {
+  // Whether we have been notified of a pending update from the watcher. This is
+  // set synchronously when any dependencies of the compute function change.
+  #isNotified = false;
+  #state = new Signal.State<AsyncComputedState>("initial");
+
+  /**
+   * The current state of the AsyncComputed, which is one of 'initial',
+   * 'pending', 'complete', or 'error'.
+   *
+   * The state will be 'initial' until the compute function is first run.
+   *
+   * The state will be 'pending' while the compute function is running. If the
+   * state is 'pending', the `value` and `error` properties will be the result
+   * of the previous run of the compute function.
+   *
+   * The state will be 'complete' when the compute function has completed
+   * successfully. If the state is 'complete', the `value` property will be the
+   * result of the previous run of the compute function and the `error` property
+   * will be `undefined`.
+   *
+   * The state will be 'error' when the compute function has completed with an
+   * error. If the state is 'error', the `error` property will be the error that
+   * was thrown by the previous run of the compute function and the `value`
+   * property will be `undefined`.
+   *
+   * This value is read from a signal, so any signals that read the state will
+   * be marked as dependents of it.
+   */
+  get state() {
```

Comment:
I think maybe this should be named `status`, since state is a type of signal too.